### ISSUE TITLE
Evaluating a method call with an out var parameter in Watch Window is painful 

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#16062]
Watch Window doesn't provide good experience for evaluating expressions like ```F(out var x)```.

A couple of problems:
1) The variable is declared in a global scope every time the expression is evaluated. That effectively means the expression can be evaluated only once, since the next evaluation reports an error that the variable has already been defined.
2) The newly introduced variable is added to Locals list, polluting it for the entire debug session, not just for the current method frame. If the active frame is switched to a method that defines a variable of the same name the Locals window now has two variables of the same name and there is no distinction of which one is which.
3) In order to display the value of ```x``` one needs to either add another expression (```x```) to the watch window or use a tuple trick and evaluate ```(F(out var x), x)```. 

[1] is a big pain, [2] and [3] are inconveniences. 

We should try to find a better approach. 
Scoping the variables to the expression itself would address [1], but would make [2] worse since it won't be possible to see the variable outside the expression thus making the tuple trick a necessity. However, we could present the result as an expandable node with the root value being the value of the expression itself and a row for each out var defined in the expression. That would address all the issues above. 

If the user actually wanted to access the value of the out variable from another expression they could create an object ID for the row displaying the variable value. We could also provide another context menu command that would elevate the variable to globals.

**Comment by gafter** [comment]
It would be nice to have this fixed sooner than "Unknown", as it prevents using anything but the simplest pattern-matching in conditional breakpoints.
